client/src/app/shared/misc/list-overflow.component.ts

```diff
@@ -0,0 +1,114 @@
+import {
+  Component,
+  Input,
+  TemplateRef,
+  ViewChildren,
+  ViewChild,
+  QueryList,
+  ChangeDetectionStrategy,
+  ElementRef,
+  ChangeDetectorRef
+} from '@angular/core'
+import { ɵBrowserDomAdapter } from '@angular/platform-browser'
+import { NgbModal, NgbDropdown } from '@ng-bootstrap/ng-bootstrap'
+import { uniqueId, lowerFirst } from 'lodash-es'
+import { ScreenService } from './screen.service'
+import { take } from 'rxjs/operators'
+
+export interface ListOverflowItemComponent {
+  label: string
+  routerLink: any
+}
+
+@Component({
+  selector: 'list-overflow',
+  templateUrl: './list-overflow.component.html',
+  styleUrls: [ './list-overflow.component.scss' ],
+  changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class ListOverflowComponent<T extends ListOverflowItemComponent> {
+  @ViewChild('modal', { static: true }) modal: ElementRef
+  @ViewChild('itemsParent', { static: true }) itemsParent: any
+  @ViewChildren('itemsRendered') itemsRendered: QueryList<HTMLElement>
+  @Input() items: T[]
+  @Input() itemTemplate: TemplateRef<{item: any}>
+
+  dom: ɵBrowserDomAdapter
+  showItemsUntilIndexExcluded: number
+  active = false
+  isInTouchScreen = false
+  isInMobileView = false
+
+  private openedOnHover = false
+
+  constructor (
+    private cdr: ChangeDetectorRef,
+    private modalService: NgbModal,
+    private screenService: ScreenService
+  ) {
+    this.dom = new ɵBrowserDomAdapter()
+  }
+
+  get showMenu () {
+    return this.showItemsUntilIndexExcluded
+  }
+
+  onListResized () {
+    this.isInTouchScreen = !!this.screenService.isInTouchScreen()
+    this.isInMobileView = !!this.screenService.isInMobileView()
+
+    const getWidth = (i: number) => document.getElementById(this.getId(i)).getBoundingClientRect().width
+    let showItemsUntilIndexExcluded: number
+    let accWidth = 0
+
+    for (const [index, el] of this.itemsRendered.toArray().entries()) {
+      accWidth += getWidth(index)
+      if (!showItemsUntilIndexExcluded) {
```

Comment:
`showItemsUntilIndexExcluded === undefined`, so the reader knows this condition is not interested by `0`